### PR TITLE
Added audio to MatchEnd.

### PR DESCRIPTION
**Pull request by LadyEbony**
### Motivation and Context
- [X] Why is this change required? What problem does it solve?
Match needs audio when the match ends. Also, MatchEndUI was using Text UI (bad)
- [ ] If it fixes an open issue, include the text `Closes #1` (where 1 would be the issue number) to your commit message.

### Types of changes
What types of changes does your code introduce? Check all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
Match End UI uses a TextPro and AudioSource component instead of the enabling/disabling two GameObjects. 

### Description
- [x] Describe your changes in detail.
MatchEndUI uses TextPro and AudioSource, same one used by MatchCountdownUI, for its announcement. Plus some commits.

### CODE REVIEW

File: Assets/Code/src/Runtime/Match/Components/MatchEndUI.cs [hunk 1]
```diff
@@ -1,29 +1,67 @@
...
+    void Awake() {
+        Mediator.Global.CreateUnityContext(this).Subscribe<MatchEndEvent>(OnMatchEnd);
+    }
+
+    
+    async Task OnMatchEnd(MatchEndEvent evt) {
+        ObjectUtil.SetActive(TextUI.gameObject, true);
+        if (evt.MatchConfig.Time > 0 && evt.MatchState.Time <= 0) {
+            UpdateTextUI("TIME!", TimeAudioClip);
+        }
```
Comment (james7132) [inline Assets/Code/src/Runtime/Match/Components/MatchEndUI.cs]: Style: `} else {`

File: Assets/Code/src/Runtime/Match/Components/MatchEndUI.cs [hunk 2]
```diff
@@ -1,29 +1,67 @@
...
+        else {
+            UpdateTextUI("GAME!", GameAudioClip);
+        }
+        await Task.Delay((int)(PostGameWait * 1000));
+    }
+
+    void UpdateTextUI(string text, AudioClip audio)
```
Comment (james7132) [inline Assets/Code/src/Runtime/Match/Components/MatchEndUI.cs]: Style: for consistency, use K&R styling. `{` on same line.

File: Assets/Code/src/Runtime/Match/Components/MatchEndUI.cs [hunk 3]
```diff
@@ -1,29 +1,67 @@
...
+    public AudioClip TimeAudioClip;
+    public AudioClip GameAudioClip;
+
+    /// <summary>
+    /// Awake is called when the script instance is being loaded.
+    /// </summary>
+    void Awake() {
+        Mediator.Global.CreateUnityContext(this).Subscribe<MatchEndEvent>(OnMatchEnd);
+    }
+
```
Comment (james7132) [inline Assets/Code/src/Runtime/Match/Components/MatchEndUI.cs]: Style: one space between elements.

**Review by james7132 — APPROVED** [review]
Mostly style nitpicks. Please change the parts mentioned on Discord.

**Comment by LadyEbony** [comment]
Addressed code style.

**Pull request merged**